Add a QB primer (#8780)
geldata/gel: docs/reference/using/python/api/fastapi.rst

.. _gel-python-fastapi:

=============
FastAPI + Gel
=============

.. py:currentmodule:: gel

> The ``gel-python`` package offers a sleek addition to FastAPI. With the built-in tools for schema migration management and automatic reflection into Pydantic models you can quickly integrate into the FastAPI development workflow. The ORM and query-building capability of Gel make building endpoints. So install ``gel`` and ``fastapi`` Python packages, run :gelcmd:`init` to setup the database and let's get started with this primer.

Reviewer: > The ORM and query-building capability of Gel make building endpoints

Make building endpoints what?

> and let's get started with this primer.

This feels a bit unnecessary.

To initialize a Gel-aware FastAPI project, create the following file:

.. code-block:: python
  :caption: main.py

  import fastapi
  import gel.fastapi

  app = fastapi.FastAPI()
  db = gel.fastapi.gelify(app)

  @app.get("/")

>   async def index():
>       # get the Gel client associated with the app
>       db = g.client
>       # This is an actual (trivial) query sent to Gel
>       return await db.query_single("select 'Hello world!'")

Reviewer: I'm a little confused by this. Shouldn't it be:

```suggestion
  async def index(client: gel.fastapi.Client):
      # This is an actual (trivial) query sent to Gel
      return await client.query_single("select 'Hello world!'")
```
